### ISSUE TITLE
Better display of Peertube video in mastodon clients

### ISSUE THREAD

**Issue by Aldarone** [tootsuite/mastodon#6983]
Following Chocobozzz/PeerTube#453, I suggest that Mastodon displays the thumbnail of Peertube videos as if it where a toot with an embed media instead of just the video title + the link.

This would feel way more engaging.

**Comment by ThibG** [comment]
I don't think that alone would make it much better, as you wouldn't have the embedded player at all anymore… maybe special handling that would display the preview icon and load the embedded player when you click on it? But that would require more work…

I think there is also a wider issue with preview cards. It is a bit weird that they are only displayed in detailed view, without any sign that there is a preview card when seen in a TL.

**Comment by rugk** [comment]
> I think there is also a wider issue with preview cards. It is a bit weird that they are only displayed in detailed view, without any sign that there is a preview card when seen in a TL.

Better open a new issue for that, can you?

**Comment by yassinphilip** [comment]
If you mean that the following are two different issues:

- There should be some indicator that a preview is "available"
- There should be some sort of preview at all

I humbly think they are not, the second invalidating the first.

**Comment by rugk** [comment]
No the difference is:
* Something should be done about Peertube video display. (i.e. some image shown)
* There should be some sort of preview at all.

If you think that the first one necessarily involves the second one, then this issue can of course be re-titled, but it is not specific to PeerTube then, anymore.

**Comment by yassinphilip** [comment]
Ah, I see what you mean. And yes, I think this issue should be renamed, and that it's not specific to peertube (I just checked a bunch of sites previews).

**Comment by mal0ki** [comment]
Going to label this as an AP enhancement (even though it may not be _only_ about AP, but about compatibility with other AP implementations to make a smooth relationship, does that make sense?

**Comment by Cambridgeport90** [comment]
What is happening with this so far?

**Comment by Tak** [comment]
Annual status query?

**Comment by ThibG** [comment]
Peertube videos do generate a preview card, which is shown in timelines as well as detailed toot view.

However, the preview isn't generated immediately.

**Comment by Tak** [comment]
Oops, you're right, it just didn't show up as quickly as I had expected
